rust/processor/src/processors/token_processor.rs

```diff
@@ -889,9 +901,14 @@ impl ProcessorTrait for TokenTransactionProcessor {
             all_current_token_claims.extend(current_token_claims);
 
             // ANS lookups
-            let current_ans_lookups =
-                CurrentAnsLookup::from_transaction(txn, self.ans_contract_address.clone());
-            all_current_ans_lookups.extend(current_ans_lookups);
+            let current_ans_lookups = CurrentAnsLookup::from_transaction(
+                txn,
+                &mut address_to_primary_name_records,
+                self.ans_primary_names_table_handle.clone(),
+                self.ans_name_records_table_handle.clone(),
+                &mut conn,
+            );
+            all_current_ans_lookups.extend(current_ans_lookups.clone());
```

Comment:
I think that you still need to pass `all_current_ans_lookups` back to the function to handle the case that same batch may have data you need.

Reply:
The current batch tracks address -> name mappings in `address_to_primary_name_records`